crypto/src/dex/swap/plaintext.rs

```diff
@@ -49,6 +52,38 @@ impl SwapPlaintext {
         )
     }
 
+    pub fn output_notes(&self, batch_data: &BatchSwapOutputData) -> (Note, Note) {
+        let (output_1_blinding, output_2_blinding) = self.output_blinding_factors();
+
+        let (lambda_1_i, lambda_2_i) = batch_data.pro_rata_outputs((
+            // TODO: fix up amount conversions
+            self.delta_1_i.try_into().unwrap(),
+            self.delta_2_i.try_into().unwrap(),
```

Comment:
Here and elsewhere there are a lot of issues where we do internal math on `u64` values, and bridging that with an `Amount` type is difficult. I think this speaks to some limitations of the existing `Amount` type we should address, in particular the lack of an analogue of a "widening multiply + divide" (which we can do for `u64` by means of `u128`).

We should fix those limitations and use `Amount` everywhere before trying to push forward on 128-bit `Amount`s.